ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/discovery/internal/IndirectVertxTimerUtil.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.p2p.discovery.internal;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+
+import io.vertx.core.Vertx;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/*
+ * In May of 2020 we had issues with some combination of vertx, circleci, and
+ * the machines it was running on no respecting timer delays.  The solution was
+ * to use JDK executors for those timer delays.
+ *
+ * This should be revisited at a future date to see if we still need this.
+ */
+public class IndirectVertxTimerUtil implements TimerUtil {
+
+  private final ScheduledExecutorService secheduledExecutor =
+      Executors.newSingleThreadScheduledExecutor();
+  private final AtomicLong nextId = new AtomicLong(0);
+  private final Map<Long, ScheduledFuture<?>> timers = new HashMap<>();
+  private final Vertx vertx;
+
+  public IndirectVertxTimerUtil(final Vertx vertx) {
+    this.vertx = vertx;
+  }
+
+  @Override
+  public long setPeriodic(final long delayInMs, final TimerHandler handler) {
+    final long id = nextId.get();
+    timers.put(
+        id,
+        secheduledExecutor.scheduleAtFixedRate(
+            () -> vertx.executeBlocking(e -> handler.handle(), r -> {}),
```

Comment:
One issue to investigate here is whether this will cause threading issues.  IIRC, `VertexTimerUtil` guarantees the timed callbacks happen in the calling thread, and I _think_ we need that as the code is currently written ...

Reply:
That would be the event loop or io thread in this case.  Doing cryptography in the event loop or io thread could be a limiter to the number of concurrent peers we can service, especially when it could block on randomness.  

While I haven't proven it yet I think this (too much processing, like walking a 4000 entry hash table) is what's causing our eth/65 issues, so getting stuff off of the io thread looks to be a long term goal.

Reply:
That makes sense.  We just need to make sure that the code we're executing with this timer is thread-safe.